src/service.py

```diff
@@ -397,6 +397,13 @@ def redfish_conn_params_valid(self, redfish_conn_params: Dict[str, str]) -> Opti
         if not redfish_conn_params:
             return None
 
+        # Skip redfish validation if either username/password is empty.
+        if redfish_conn_params and not (
```

Comment:
The previous statement already checks for presence of redfish_conn_params.

```suggestion
        if not (
```